jobs/FindLabeledImagesInResults.php

```diff
@@ -0,0 +1,171 @@
+            $this->log( curl_error( $this->ch ) . ':' . curl_errno( $this->ch ) );
+            die( 'Exiting because of curl error, see log for details.' );
+        }
+        $array = json_decode( $result, true );
+        return $array;
+    }
+
+    private function getSearchUrl( string $searchTerm ) : string {
+        return sprintf(
+            $this->searchUrl . '/w/index.php?search=%s+filetype:bitmap&ns6=1&' .
+            'cirrusDumpResult&mediasearch=1&limit=100',
+            urlencode( $searchTerm )
+        );
+    }
+
+    private function processResults( string $searchTerm, array $searchResults, int $searchId ) {
+        $labeledData = $this->getLabeledData( $searchTerm );
+
+        $titles = [];
+        if ( isset( $searchResults['__main__']['result']['hits']['hits'] ) ) {
+            foreach ( $searchResults['__main__']['result']['hits']['hits'] as $index => $result ) {
+                $titles[] = $this->extractTitle( $result['_source'] );
+            }
+        }
+        $this->log( 'Found ' . count( $titles ) . ' results' );
+        if ( count( $titles ) > 0) {
+            $this->db->query(
+            'insert into resultset set ' .
+                'searchId=' . intval( $searchId ) . ', ' .
+                'term="' .  $this->db->real_escape_string( $searchTerm ) . '", ' .
+                'resultCount=' . intval( count( $titles ) )
+            );
+            $resultsetId = $this->db->insert_id;
+            $labeledImageCount = 0;
+            foreach ( $titles as $index => $title ) {
+                if ( isset( $labeledData[$title] ) ) {
+                    $this->db->query(
+                        'insert into labeledResult set ' .
+                        'resultsetId=' . intval( $resultsetId ) . ', ' .
+                        'filePage="' .  $this->db->real_escape_string( $title ). '", ' .
+                        'position=' . intval( $index ) . ', ' .
+                        'rating=' . intval( $labeledData[$title] )
+                    );
+                    $labeledImageCount++;
+                }
+            }
+            $this->log( $labeledImageCount . ' of the results were labeled' );
+        } else {
+            $this->db->query(
+                'insert into resultset set ' .
+                'searchId=' . intval( $searchId ) . ', ' .
+                'term="' .  $this->db->real_escape_string( $searchTerm ) . '", ' .
+                'resultCount=0'
+            );
+        }
+    }
+
+    private function getLabeledData( string $searchTerm ) : array {
+        $return = [];
+        $labeledImages = $this->db->query(
+            'select file_page,rating from results_by_component where
+            term="' . $this->db->real_escape_string( $searchTerm ) .'"
+            and rating is not null'
+        );
+        while ( $labeledImage = $labeledImages->fetch_assoc() ) {
+            $return[$labeledImage['file_page']] = $labeledImage['rating'];
```

Comment:
This might need a check for conflicting labels - there are a few cases where the same page received a different rating